### PR TITLE
Scripting: add access to AC_PID objects

### PR DESCRIPTION
**Pull request by IamPete1**
This adds access to the main AC_PID objects in each of the vehicles. This allows getting and setting of gains and could pave the way for a scripting auto tune. 

There are a couple more AC_PID objects in the the code, winch, wheel speed maybe some others.

Tested with included demo script on all vehicles.

### CODE REVIEW

**Comment by WickedShell** [comment]
Except for the reset's what the advantage here over just doing param manipulation?

**Comment by IamPete1** [comment]
@WickedShell speed, probably a bad idea to be using get by name at 100hz, I did make an attempt to get params objects but I'm failing

**Comment by WickedShell** [comment]
You're tweaking your tuning PID's at 100's of Hz?

File: libraries/AP_Vehicle/AP_Vehicle.h [hunk 1]
```diff
@@ -74,6 +75,21 @@ class AP_Vehicle : public AP_HAL::HAL::Callbacks {
...
+        RATE_YAW = 2,
+        RATE_STEER = 3,
+        THROTTLE_SPEED = 4,
+        PITCH_THROTTLE = 5,
+        SAILBOAT_HEEL = 6,
+        TRACKER_PITCH = 7,
+        TRACKER_YAW = 8,
+    };
+
+    virtual AC_PID* get_AC_PID(AC_PID_TYPE type) {return nullptr;};
```
Comment (rmackay9) [inline libraries/AP_Vehicle/AP_Vehicle.h:91]: nitpick: think there should be a couple of spaces after/before the curly brackets.

**Comment by rmackay9** [comment]
I'm generally OK with this change.  I think the rename of the AC_PID object methods is an improvement.

My only firm request is that we try to keep the placement of the AP_Vehicle override methods tidy.  The existing methods, "start_takeoff", "set_target_location" and "set_target_velocity_NED" but the new on is in system.cpp.  I think we should consider putting them all together somewhere.  so that could either be an AP_Vehicle.cpp or Copter.cpp or some new file.

**Comment by IamPete1** [comment]
This was discussed on the dev-call, general consensus was that this is better done via some sort of direct parameter access

**Pull request closed without merging**